[quant][graphmode] Add e2e test for conv+bn

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #27403 [quant][graphmode] Minor fix and turn off fold_convbn
* **#27348 [quant][graphmode] Add e2e test for conv+bn**

Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D18182920](https://our.internmc.facebook.com/intern/diff/D18182920)

jerryzh168 committed:
[quant][graphmode] Add e2e test for conv+bn

Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add e2e test for conv+bn"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add e2e test for conv+bn"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add e2e test for conv+bn"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add e2e test for conv+bn"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add e2e test for conv+bn"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add e2e test for conv+bn"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add e2e test for conv+bn"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

ZolotukhinM commented on test/test_quantization.py:
```diff
@@ -699,6 +698,34 @@ def test_single_layer(self):
+            [self.img_data],
+            inplace=False)
+        result_eager = model_eager(self.img_data[0][0])
+        result_script = get_forward(model_script._c)(self.img_data[0][0])
```
You could call it the same way you call the eager model:
```
result_script = model_script(self.img_data[0][0])
```

ZolotukhinM commented on test/test_quantization.py:
```diff
@@ -699,6 +698,34 @@ def test_single_layer(self):
+        qconfig_dict = {
+            '': default_qconfig
+        }
+        model_script = quantize_script(
```
We need to somehow verify that we're actually running quantized ops. If for some reason fusion doesn't work, this test won't catch it.

jerryzh168 replied:
do you have similar tests for graph executor?

ZolotukhinM reviewed (approved):
This test would not catch if fusion (or any of other "optional" passes) would stop working, but it's still better to have it. Please think about how we can test passes that can silently do nothing.

jerryzh168 committed:
Update on "[quant][graphmode] Add e2e test for conv+bn"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add e2e test for conv+bn"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add e2e test for conv+bn"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add e2e test for conv+bn"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add e2e test for conv+bn"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add e2e test for conv+bn"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add e2e test for conv+bn"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add e2e test for conv+bn"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add e2e test for conv+bn"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add e2e test for conv+bn"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add e2e test for conv+bn"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add e2e test for conv+bn"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add e2e test for conv+bn"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add e2e test for conv+bn"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add e2e test for conv+bn"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add e2e test for conv+bn"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D18182920](https://our.internmc.facebook.com/intern/diff/D18182920)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add e2e test for conv+bn"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D18182920](https://our.internmc.facebook.com/intern/diff/D18182920)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add e2e test for conv+bn"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D18182920](https://our.internmc.facebook.com/intern/diff/D18182920)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add e2e test for conv+bn"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D18182920](https://our.internmc.facebook.com/intern/diff/D18182920)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add e2e test for conv+bn"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D18182920](https://our.internmc.facebook.com/intern/diff/D18182920)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add e2e test for conv+bn"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D18182920](https://our.internmc.facebook.com/intern/diff/D18182920)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add e2e test for conv+bn"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D18182920](https://our.internmc.facebook.com/intern/diff/D18182920)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add e2e test for conv+bn"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D18182920](https://our.internmc.facebook.com/intern/diff/D18182920)

[ghstack-poisoned]
